.gitignore

```diff
@@ -37,3 +37,12 @@ tags
 
 # VSCode specific
 .vscode
+# Ignore everything in vendor except github.com/knative/test-infra
```

Comment:
can we just move github.com/knative/test-infra into a separate directory (not vendor). .gitignore looks pretty weird.

Reply:
This not only includes the shell script, it includes a little bit of go code, which has to be on an appropriate path. I'll file a bug on the test-infra repo to support modules better, but this was the best compromise I could think of between the module way and the vendor way. This way, `git mod vendor` does the right thing to the repo, but we don't check in all the vendored deps.

Reply:
I already filed https://github.com/knative/test-infra/issues/546, feel free to add more details since you know firsthand the limitations.